app/Http/Controllers/Tool2Controller.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Controllers\Controller;
+use App\Models\Role;
+use App\Models\Tool;
+use App\Models\User;
+use Illuminate\Http\Request;
+
+class Tool2Controller extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Contracts\Foundation\Application|\Illuminate\Contracts\View\Factory|\Illuminate\Contracts\View\View
+     */
+    public function index()
+    {
+        $tools = Tool::all();
+        return view('tools.index', ['tools' => $tools]);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Contracts\Foundation\Application|\Illuminate\Contracts\View\Factory|\Illuminate\Contracts\View\View
+     */
+    public function create()
+    {
+        $max_id= Tool::max('id');
+        $counter = $max_id +1;
+
+        $tools = Tool::all();
+
+
+        return view('tools.create', ['counter' =>$counter, 'tools' => $tools]);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function store(Request $request)
+    {
+        $request->validate([
+            'name' => 'required',
+            'status' => 'required',
+            'tutorial' => 'required'
```

Comment:
Hier sollte die Länge der Eingabe beschränkt werden. Maximal auf die Länge die in der Datenbank erlaubt ist. (Meist 255)